src/app/current-users/current-users.component.ts

```diff
@@ -12,64 +13,73 @@ import { UsersObject } from '../../../api-objects/UsersObject';
 export class CurrentUsersComponent implements OnInit {
   public usersTable: UsersObject[] = [];
   public pageCounter: number = 0;
+  public authorityTypes = authorityTypes;
 
-  constructor(/*private currentusersTableService: GetCurrentUsersService*/) { }
+  constructor(private apiUsersCrudService: ApiUsersCrudService, private notificationService: NotificationService) { }
 
   public ngOnInit() {
     this.getAllUsers();
-}
-
- public getAllUsers(): void {
-  //  this.currentUsersService.getUsers()
-  //  .subscribe(
-  //    (results) => {
-  //      this.usersTable = results.data;
-  //      console.log(this.usersTable);
-  //      console.log('Users loading...');
+  }
 
-  //  },(error) => {
-  //    console.log('There was an error' + error);
-  //  });
+  public getAllUsers(): void {
+    this.apiUsersCrudService.getUsers()
+      .subscribe(
+        (results) => {
+          this.usersTable = results.data;
+        });
+  }
 
-  this.usersTable = Users;
-  // this.usersTable.map((types) => {
-  //   if(types.authorityType === 0) {} else {}
-  // });
- }
+  public prevPage(pageNum: number): void {
+    while (this.pageCounter > 0 && pageNum > 0) {
+      this.pageCounter -= 1;
+      pageNum -= 1;
+    }
+  }
 
- public prevPage(pageNum: number): void {
-   while( this.pageCounter > 0 && pageNum > 0) {
-     this.pageCounter -= 1;
-     pageNum -= 1;
-   }
- }
+  public nextPage(pageNum: number): void {
+    while ((this.pageCounter + 1) * 10 < this.usersTable.length && pageNum > 0) {
+      this.pageCounter += 1;
+      pageNum += 1;
+    }
+  }
 
- public nextPage(pageNum: number): void {
-   while((this.pageCounter + 1) * 10 < this.usersTable.length && pageNum > 0) {
-     this.pageCounter += 1;
-     pageNum += 1;
-   }
- }
+  /**
+   * Deletes a user at a given index.
+   * @param user a user object to be deleted
+   */
+  public deleteUser(user: UsersObject) {
+    this.apiUsersCrudService.deleteUser(user.username).subscribe(
```

Comment:
This should pass in an authortiyId.
```suggestion
    this.apiUsersCrudService.deleteUser(user.authorityId).subscribe(
```